### ISSUE TITLE
Add the support for tensor network benchmarking

### ISSUE THREAD

**Issue by liweintu** [qiboteam/qibojit-benchmarks#41]
We add the support for the benchmarking of tensor network backend in simulation. First quimb, then cuQuantum.

**Comment by liweintu** [comment]
To add qibotn as a backend, there're two options.
1. In `.benchmarks/libraries/qibo.py`, extend the Qibo class to support `qibotn` as a new backend.
2. In `.benchmarks/libraries/`, create a new file `qibotn.py` to add qibo-supported TNet backends.

IMHO, if we want to package both state vector and TNet support into qibo, option 1 is favoured.

@AleCandido Which one do you recommend?

**Comment by AleCandido** [comment]
Given that `qibotn` is somehow falling under the hat of Qibo, I would see option 1 as the favored alternative as well, exactly for the same reason you are arguing.

However, in this context I'm not an authoritative source, I believe we should ask for the opinion of @stavros11 (who mostly developed the benchmarks and a large part of Qibo itself), and of course @scarrazza.

**Comment by liweintu** [comment]
> Given that `qibotn` is somehow falling under the hat of Qibo, I would see option 1 as the favored alternative as well, exactly for the same reason you are arguing.
> 
> However, in this context I'm not an authoritative source, I believe we should ask for the opinion of @stavros11 (who mostly developed the benchmarks and a large part of Qibo itself), and of course @scarrazza.

Sure, fair enough.

Hi @stavros11 , @scarrazza , do you have a suggestion about which option we should pick to incorporate TNet backends into the benchmarking? Thanks.

**Comment by liweintu** [comment]
Agreed, then we'll go ahead to follow `qibojit` backend as the template. Thanks for the suggestion @stavros11 .
